tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ContextPropagation.cs

```diff
@@ -8,28 +8,33 @@
 using System.Collections.Generic;
 using System.Text;
 using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Headers;
 using Datadog.Trace.Propagators;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
 
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SQS
 {
     internal static class ContextPropagation
     {
-        private const string SqsKey = "_datadog";
+        internal const string SqsKey = "_datadog";
 
-        private static void Inject<TMessageRequest>(SpanContext context, IDictionary messageAttributes)
+        private static void Inject<TMessageRequest>(SpanContext context, IDictionary messageAttributes, DataStreamsManager dataStreamsManager)
         {
             // Consolidate headers into one JSON object with <header_name>:<value>
             var sb = Util.StringBuilderCache.Acquire(Util.StringBuilderCache.MaxBuilderSize);
             sb.Append('{');
             SpanContextPropagator.Instance.Inject(context, sb, default(StringBuilderCarrierSetter));
+            dataStreamsManager.InjectPathwayContext(context.PathwayContext, new StringBuilderJsonAdapter(sb));
```

Comment:
The `dataStreamsManager` can be `null`, so if you want to allow passing null then you can use this suggested null-check

```suggestion
            dataStreamsManager?.InjectPathwayContext(context.PathwayContext, new StringBuilderJsonAdapter(sb));
```

Reply:
thanks for the tip